### ISSUE TITLE
Custom .env file via parsed argument fails

### ISSUE THREAD

**Issue by mjbear** [theskumar/python-dotenv#327]
I am writing a script that can run against different dotenv config files based on what is passed to the script as an argument. Although the dotenv file name is picked up from argparse it is unfortunately not loaded. While manually hard coding the file name (whether to load_dotenv() or as an "in code" variable) does load as expected, but it isn't flexible/reusable code then.

I did find a work around of specifying the **type=str** (to argparse's ArgumentParser.add_argument method) instead of type=ascii. I have included examples below.

Dotenv Example:
```
USERNAME = bobby
PASSWORD = password123
```

Python Example:
```
import os
import argparse
from dotenv import load_dotenv, dotenv_values

argp = argparse.ArgumentParser(
        description='a test script'
)

argp.add_argument('-c', '--config', metavar='DOTENV-CONFIG', type=ascii,
                  required=True, help='specify a dotenv configuration file')
                  
# setting the type= to str instead appears to be a workaround
# argp.add_argument('-c', '--config', metavar='DOTENV-CONFIG', type=str,
#                   required=True, help='specify a dotenv configuration file')

args = vars(argp.parse_args(['-c', '.env-blah']))

config_file = args['config']
print(f'Config File: {config_file} of type {type(config_file)}')


# parsed file name from argument fails to load
##
# load_dotenv(config_file, override=True, verbose=True)
# load_dotenv(config_file, override=False, verbose=True)
load_dotenv(config_file)
# load_dotenv(config_file.encode('ascii'))

print(os.environ)

print('=' * 30)


# Static File Name
# - less flexible to specify a static dotenv file name, but it does work
# - loads additional environment variables
##
config_file2 = '.env-blah'
load_dotenv(config_file2)
# load_dotenv(config_file2.encode('ascii'))

print(os.environ)
```

**Comment by mjbear** [comment]
@theskumar , @bbc2 

Probably not related to #321 as my dotenv file exists, but is similar to #274

Thank you!

**Comment by bbc2** [comment]
What's the problem with `type=str`?  It looks to me like `type=ascii` doesn't make sense since it will surround your string with single quotes and thus turn it into an invalid path.

**Comment by mjbear** [comment]
There isn't a problem with `type=str`.

When I browsed through the doc page on argparse the supported [https://docs.python.org/3/library/argparse.html#type](add_argument object types and examples) didn't show str being an option. Now that I see the "user defined functions" example it's clearer that just about any object type may be possible (and certainly strings as I found out).

Apologies for misinterpreting a gap in argparse documentation to a possible issue with python-dotenv.
